tests/testDeviceAccessLib.py

```diff
@@ -0,0 +1,247 @@
+        readout = da.getDMapFilePath()
+        self.assertEqual(readout, self.dmap_filepath)
+
+    def testReads_Register_Name(self):
+        readout = self.twoD_in32_acc.getName()
+
+        self.assertEqual(readout, self.int_2d_int32_acc_name)
+
+    def testRead_Unit_from_Acc(self):
+        readout_unset = self.twoD_in32_acc.getUnit()  # unit is not set
+        self.assertEqual(readout_unset, "n./a.")
+        # Other values cannot be set. Has to be set via custom backend to be tested
+
+    def testCheck_Access_Mode_Flags(self):
+        # standard flags are empty:
+        empty_flags = self.twoD_in32_acc.getAccessModeFlags()
+        self.assertEqual([], empty_flags)
+
+        # wait_for_new_data:
+        wait_for_new_data_flag = self.async_read_1d_int32_acc.getAccessModeFlags()
+        self.assertEqual([da.AccessMode.wait_for_new_data], wait_for_new_data_flag)
+
+        # check raw
+        raw_data_flag = self.raw_1d_int32_acc.getAccessModeFlags()
+        self.assertEqual([da.AccessMode.raw], raw_data_flag)
+
+        # check both
+        raw_and_async_data_flags = self.async_raw_1d_int32_acc.getAccessModeFlags()
+        self.assertIn(da.AccessMode.raw, raw_and_async_data_flags)
+        self.assertIn(da.AccessMode.wait_for_new_data, raw_and_async_data_flags)
+
+    def testReadWriteFlags(self):
+        self.assertTrue(self.rw_int32_acc.isReadable())
+        self.assertTrue(self.ro_int32_acc.isReadable())
+        self.assertFalse(self.wo_int32_acc.isReadable())
+
+        self.assertTrue(self.rw_int32_acc.isWriteable())
+        self.assertFalse(self.ro_int32_acc.isWriteable())
+        self.assertTrue(self.wo_int32_acc.isWriteable())
+
+        self.assertFalse(self.rw_int32_acc.isReadOnly())
+        self.assertTrue(self.ro_int32_acc.isReadOnly())
+        self.assertFalse(self.wo_int32_acc.isReadOnly())
+
+    def testReadLatest(self):
+        # pay attention to use a normal dummy, shared memory dummy will have a race condition that will fail the test
+
+        # discard first, non-blocking read
+        self.async_scalar_push.read()
+
+        # fill the queue with 1 to 10 (10 values):
+        for i in range(1, 11):
+            self.write_scalar.set(i)
+            self.write_scalar.write()
+            self.scalar_data_ready.write()
+
+        # queue is full:
+        new_value_available = self.async_scalar_push.readLatest()
+        self.assertTrue(new_value_available)
+        result = self.async_scalar_push
+        self.assertEqual(result, 10)
+
+        # queue is empty:
+        new_value_available = self.async_scalar_push.readLatest()
+        self.assertFalse(new_value_available)
+
+    def testReadNonBlocking(self):
+        # pay attention to use a normal dummy, shared memory dummy will have a race condition that will fail the test
+        some_value = 445
+
+        # discard first, non-blocking read
+        self.async_scalar_push.read()
+
+        # put one in the queue
```

Comment:
To discriminate from readLatest, you should put two into the queue and make sure that readNonBlocking will read both of them (and not just the second)

Reply:
Good catch! I will add that.